File: RenetikObjc/Library/Classes/Core/Categories/UIKit/UISegmentedControl+CSExtension.m
//
// Created by Rene Dohan on 26/05/18.
// Copyright (c) 2018 Renetik Software. All rights reserved.
//

#import "UISegmentedControl+CSExtension.h"

@implementation UISegmentedControl (CSExtension)
- (NSString *)selectedTitle {
    return self.selectedSegmentIndex != UISegmentedControlNoSegment ?
            [self titleForSegmentAtIndex:(NSUInteger) self.selectedSegmentIndex] : nil;
}

+ (int)DEFAULT_HEIGHT {
    return 28;
}

- (NSInteger)selectedIndex {
    return self.selectedSegmentIndex;
}

- (void)setSelectedIndex:(NSInteger)index {
    self.selectedSegmentIndex = index;
}
@end

File: Sources/RenetikCore/extensions/UIKit/UISegmentedControl+CSExtension.swift
//
// Created by Rene Dohan on 3/4/20.
//

import UIKit

public extension UISegmentedControl {
    class var defaultHeight: CGFloat { 28 }

    @discardableResult
    func selected(index: Int) -> Self { self.selectedIndex = index; return self }

    var selectedIndex: Int {
        get { selectedSegmentIndex }
        set { selectedSegmentIndex = newValue }
    }

    var selectedTitle: String? {
        selectedSegmentIndex != UISegmentedControl.noSegment ?
                titleForSegment(at: selectedSegmentIndex) : nil
    }


}
